src/regs.rs

```diff
@@ -93,12 +93,12 @@ impl GeneralPurposeRegisters {
     }
 
     /// Sets the value of the given register.
-    pub fn set_reg(&mut self, reg_index: GprIndex, val: usize) {
-        if reg_index == GprIndex::Zero {
+    pub fn set_reg(&mut self, reg_index: usize, val: usize) {

```

Comment:
why do you want to replace `GprIndex` with `usize`

It seems that `GprIndex` is better here, since you need to do so many modifications in the following `set_reg` related  functions.

Reply:
Because Zecheng added a new set_gpr function in the axvcpu trait, the parameter type is usize

Reply:
I understant that, but that is a trait implemented for `RISCVVcpu`, you can still keep methods for GeneralPurposeRegisters as it is.

Reply:
OK，but there are still many “as usize” in vcpu.rs beacuse of set_gpr

Reply:
> OK，but there are still many “as usize” in vcpu.rs beacuse of set_gpr

why not use `self.xxx.set_gpr`

Reply:
add a helper function for that then?

Reply:
add a function, such as set_gpr_from_index, to change the original set_gpr usize parameter to the GprIndex type?